crates/sol-types/src/types/event/mod.rs

```diff
@@ -151,4 +151,9 @@ pub trait SolEvent: Sized {
         let body = Self::decode_data(data, validate)?;
         Ok(Self::new(topics, body))
     }
+
```

Comment:
hot take. We should consider `#[doc(hide)]` all the trait methods that do _not_ operate on `&Log` or output `Log`, as users should generally not use them directly

Reply:
They should because `Log` can't borrow data while the other can